pkg/devfile/header.go

```diff
@@ -43,6 +43,10 @@ type DevfileMetadata struct {
 	// +optional
 	Tags []string `json:"tags,omitempty"`
 
+	// Optional list of processor architectures that the devfile supports, empty list suggests that the devfile supports all the architectures
```

Comment:
> empty list suggests that the devfile supports all the architectures

I'm not 100% sure about this, but I'm not against it as well.

I would say empty list suggests maintainer does not care about multiarch, or just misses the value. Assuming that devfile supports all the possible architecture is not right thing to do, since multiarch support is not out-of-the-box thing, but vice versa, it requires from author to make sure that all used images support all the architectures.

And what is `all architectures`? `linux/amd64,linux/arm64,linux/ppc64le,linux/s390x,linux/386,linux/arm/v7,linux/arm/v6` ?

`amd64, arm32v5, arm32v6, arm32v7, arm64v8, i386, ppc64le, and s390` ?

Reply:
@sleshchenko We listed here https://github.com/devfile/api/issues/244#issuecomment-852157289 - the best practice for writing a devfile is to have a list of all the supported architectures

if there is no architecture in the devfile, I would assume one would/could try to use the devfile for any architecture. Take our registry for example, unless explicitly specified; can one not use it for other archs?

This is not mandatory field but during registry build or on boarding of the devfile we see that the arch field is empty, we can flag it for the author's attention, as suggested earlier that it is best practice. (Let me think more about this, and I'll comment here if I have other reservations)

Arch field can incl "amd64", "arm", "386", "mips64le", "ppc64le", "s390x" (I parsed this from a docker image that supported multiple arch)

Reply:
@sleshchenko could you pls help re-review the latest push? I've updated the description to be more concise.

For now, I've used skopeo to get the architecture values from a `hello-world` image manifest. I've also added some validation for this field and only pre-selected some of the more common architectures for now that can be used(I've excluded 32bits and mips64le). 

I've not taken variants into consideration atm.